packager/packager.h

```diff
@@ -0,0 +1,86 @@
+#ifndef SORBET_REWRITER_PACKAGE_H
+#define SORBET_REWRITER_PACKAGE_H
+#include "ast/ast.h"
+
+namespace sorbet {
+class WorkerPool;
+}
+
+namespace sorbet::packager {
+/**
+ * This pass transforms package files (`foo/__package.rb`) from
+ *
+ *   class Project::Foo < PackageSpec
+ *    import Project::Bar
+ *
+ *    export Baz
+ *    export_methods FooClassWithMethods
+ *   end
+ *
+ * given
+ *
+ *   class Project::Bar < PackageSpec
+ *     export SomeClassInBar
+ *     export Inner::SomeOtherClassInBar
+ *     export_methods SomeClassWithMethods, SomeOtherClassWithMethods
+ *   end
+ *
+ * to:
+ *
+ *   module <PackageRegistry>::Project_Foo_Package
+ *     module Project::Bar
+ *       # Import methods exported on Bar
+ *       extend <PackageRegistry>::Bar_Package::<PackageMethods>
+ *       # Import each class exported by Bar
+ *       SomeClassInBar = <PackageRegistry>::Bar_Package::SomeClassInBar
+ *       SomeOtherClassInBar = <PackageRegistry>::Bar_Package::SomeOtherClassInBar
+ *     end
+ *     module <PackageMethods>
+ *       include <PackageRegistry>::Project_Foo_Package::FooClassWithMethods
+ *     end
+ *   end
+ *
+ *   class Project::Foo < PackageSpec
+ *    import Project::Bar
+ *
+ *    export <PackageRegistry>::Project_Foo_Package::Baz
+ *    export_methods <PackageRegistry>::Project_Foo_Package::FooClassWithMethods
+ *   end
+ *
+ * It also rewrites files in the package, like `foo/baz.rb`, from:
+ *
+ * class SomeOtherClassInTheFile
+ *   ...
+ * end
+ *
+ * class Baz
+ *   ...
+ * end
+ *
+ * to:
+ *
+ * module <PackageRegistry>::Project_Foo_Package
+ *   class SomeOtherClassInTheFile
+ *     ...
+ *   end
+ *
+ *   class Baz
+ *     ...
+ *   end
+ * end
+ *
+ * This pass intentionally runs outside of rewriter so that its output does not get cached.
```

Comment:
Thank you for this comment about caching.

Would you mind moving/copying this line to pipeline.cc? That's where I would have expected to find it (that's where I was thinking about caching)

Reply:
Copied it to the body of `resolve`, which calls `package`.